From c23241755f17177a506518b24aa16706cc4063ed Mon Sep 17 00:00:00 2001
From: Wael Karman <wael.karman@gmail.com>
Date: Sun, 28 Apr 2024 00:45:25 +0200
Subject: [PATCH] increase swiotlb size

---
 include/linux/swiotlb.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/linux/swiotlb.h b/include/linux/swiotlb.h
index ecde0312dd52..dbc91c9026d8 100644
--- a/include/linux/swiotlb.h
+++ b/include/linux/swiotlb.h
@@ -23,7 +23,7 @@ struct scatterlist;
  * must be a power of 2.  What is the appropriate value ?
  * The complexity of {map,unmap}_single is linearly dependent on this value.
  */
-#define IO_TLB_SEGSIZE	128
+#define IO_TLB_SEGSIZE	2048
 
 /*
  * log of the size of each IO TLB slab.  The number of slabs is command line
@@ -33,7 +33,7 @@ struct scatterlist;
 #define IO_TLB_SIZE (1 << IO_TLB_SHIFT)
 
 /* default to 64MB */
-#define IO_TLB_DEFAULT_SIZE (64UL<<20)
+#define IO_TLB_DEFAULT_SIZE (500UL<<20)
 
 unsigned long swiotlb_size_or_default(void);
 void __init swiotlb_init_remap(bool addressing_limit, unsigned int flags,
-- 
2.25.1

